nixos/modules/services/networking/frr.nix

```diff
@@ -25,210 +69,260 @@ let
-                Restart = "on-abnormal";
-              };
-            });
-       in
-         listToAttrs (map frrService (filter isEnabled allServices));
+      users.groups = {
+        frr = { };
+        # Members of the frrvty group can use vtysh to inspect the FRR daemons
+        frrvty = {
+          members = [ "frr" ];
+        };
+      };
 
-  };
+      environment.etc = {
+        "frr/frr.conf".source = configFile;
+        "frr/vtysh.conf".text = ''
+          service integrated-vtysh-config
+        '';
+        "frr/daemons".text = ''
+          # This file tells the frr package which daemons to start.
+          #
+          # Sample configurations for these daemons can be found in
+          # /usr/share/doc/frr/examples/.
+          #
+          # ATTENTION:
+          #
+          # When activating a daemon for the first time, a config file, even if it is
+          # empty, has to be present *and* be owned by the user and group "frr", else
+          # the daemon will not be started by /etc/init.d/frr. The permissions should
+          # be u=rw,g=r,o=.
+          # When using "vtysh" such a config file is also needed. It should be owned by
+          # group "frrvty" and set to ug=rw,o= though. Check /etc/pam.d/frr, too.
+          #
+          # The watchfrr, zebra and staticd daemons are always started.
+          #
+          ${daemonList}
+
+          #
+          # If this option is set the /etc/init.d/frr script automatically loads
+          # the config via "vtysh -b" when the servers are started.
+          # Check /etc/pam.d/frr if you intend to use "vtysh"!
+          #
+          vtysh_enable=yes
+          ${daemonOptions}
+
+
+          # If you want to pass a common option to all daemons, you can use the
+          # "frr_global_options" variable.
+          #
+          #frr_global_options=""
+
+
+          # The list of daemons to watch is automatically generated by the init script.
+          # This variable can be used to pass options to watchfrr that will be passed
+          # prior to the daemon list.
+          #
+          # To make watchfrr create/join the specified netns, add the the "--netns"
+          # option here. It will only have an effect in /etc/frr/<somename>/daemons, and
+          # you need to start FRR with "/usr/lib/frr/frrinit.sh start <somename>".
+          #
+          #watchfrr_options=""
+
+
+          # configuration profile
+          #
+          #frr_profile="traditional"
+          #frr_profile="datacenter"
+
+
+          # This is the maximum number of FD's that will be available.  Upon startup this
+          # is read by the control files and ulimit is called.  Uncomment and use a
+          # reasonable value for your setup if you are expecting a large number of peers
+          # in say BGP.
+          #
+          #MAX_FDS=1024
```

Comment:
systemd already limits this. Larger values than in systemd service definition will have no effect I guess. Upstream has this definition when systemd is not used I guess.

Reply:
I think I'll strip the comments from the verbatim included (upstream) `/etc/frr/daemons` template. If anyone really needs `frr_profile` or other options, we can still extend this later anyway.